app/src/lib/git/cherry-pick.ts

```diff
@@ -262,49 +287,53 @@ export async function getCherryPickSnapshot(
 
     // Each line is of the format: `pick shortSha commitSummary`
     remainingPicks.split('\n').forEach(line => {
-      const linePieces = line.split(' ')
-      if (linePieces.length > 2) {
-        remainingShas.push(linePieces[1])
+      line = line.replace(/^pick /, '')
+      if (line.trim().includes(' ')) {
+        const sha = line.substr(0, line.indexOf(' '))
+        const commit: CommitOneLine = {
+          sha,
+          summary: line.substr(sha.length + 1),
+        }
+        remainingCommits.push(commit)
       }
     })
 
-    if (remainingShas.length === 0) {
-      // This should only be possible with corrupt sequencer files.
-      return null
-    }
-    lastSha = remainingShas[remainingShas.length - 1]
-
-    if (lastSha === '') {
+    if (remainingCommits.length === 0) {
       // This should only be possible with corrupt sequencer files.
       return null
     }
   } catch {}
 
-  const commits = await getCommitsInRange(
-    repository,
-    revRangeInclusive(firstSha, lastSha)
-  )
+  // To get all the commits for the cherry-pick operation, we need to get the
+  // ones already cherry-picked. If abortSafetySha is headSha; none have been
+  // cherry-picked yet.
+  const commitsCherryPicked =
+    abortSafetySha !== headSha
+      ? await getCommitsInRange(repository, revRange(headSha, abortSafetySha))
+      : []
 
-  if (commits === null || commits.length === 0) {
+  if (commitsCherryPicked === null) {
     // This should only be possible with corrupt sequencer files resulting in a
     // bad revision range.
     return null
   }
 
-  const count = commits.length - remainingShas.length
-  const commitSummaryIndex = count > 0 ? count - 1 : 0
+  const commits = [...commitsCherryPicked, ...remainingCommits]
+  const count = commitsCherryPicked.length + 1
+
   return {
     progress: {
       kind: 'cherryPick',
       title: `Cherry-picking commit ${count} of ${commits.length} commits`,
       value: round(count / commits.length, 2),
       cherryPickCommitCount: count,
       totalCommitCount: commits.length,
-      currentCommitSummary: commits[commitSummaryIndex].summary ?? '',
+      currentCommitSummary: remainingCommits[0].summary ?? '',
     },
-    remainingCommits: commits.slice(count, commits.length),
+    remainingCommits,
     commits,
-    targetBranchUndoSha: firstSha,
+    targetBranchUndoSha: headSha,
+    countCherryPicked: commitsCherryPicked.length,
```

Comment:
Total nit, but the properties above are named `cherryPickCommitCount` and `totalCommitCount`, so maybe this one should be `cherryPickedCount`?

Reply:
Done.. and also updated the cherryPickCommitCount to position.. because it really isn't a count. It represents which commit in a selection of commits is being cherry-picked at them moment.